### PR TITLE
feat: parse diary completion from message box

### PR DESCRIPTION
**Pull request by iProdigy**
Should help with #93. Thanks to DvH for the idea

### CODE REVIEW

File: src/main/java/dinkplugin/notifiers/DiaryNotifier.java [hunk 1]
```diff
@@ -101,6 +141,9 @@ public void onVarbitChanged(VarbitChanged event) {
     }
 
     private void handle(String area, AchievementDiary.Difficulty difficulty) {
+        if (cooldownTicks.getAndSet(2) > 0)
+            return;
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/DiaryNotifier.java:145]: Maybe a little pre-emptive `log.debug` here

File: src/main/java/dinkplugin/notifiers/DiaryNotifier.java [hunk 2]
```diff
@@ -60,6 +67,39 @@ public void onTick() {
...
+            }
+
+            String area = matcher.group("area");
+            Optional<Integer> found = DIARIES.entrySet().stream()
+                .filter(e -> e.getValue().getRight() == difficulty && e.getValue().getLeft().equals(area))
+                .map(Map.Entry::getKey)
+                .findAny();
+            if (found.isPresent()) {
+                int varbitId = found.get();
+                if (isComplete(varbitId, 2)) {
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/DiaryNotifier.java:91]: This will always return true, you'll need to inverse the check to check

**Review by pajlada — CHANGES_REQUESTED** [review]
Nice, this should help with the reliability of diary alerts! Some small nitpicks

File: src/main/java/dinkplugin/notifiers/DiaryNotifier.java (hunk 2 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/notifiers/DiaryNotifier.java]: Good catch

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
Felanbird ran into an "issue" where the Western Province diary didn't match because the name in our domain file is "Western Provinces" but in the mesbox it's "Western Province"

```
2023-02-03 18:02:12 [Client] WARN dinkplugin.notifiers.DiaryNotifier - Failed to match diary area: Western Province 
```

The screenshot: https://i.ivr.fi/runelite/d31b31ebfb3f.png

A bunch of other diary completions with screenshots:
Ardougne Medium: https://i.ivr.fi/runelite/0506476b.png
Ardougne Medium: https://i.ivr.fi/runelite/3bc63bac.png
Ardougne Medium: https://i.ivr.fi/runelite/97b6029a2193.png

Desert Medium: https://i.ivr.fi/runelite/337202da248a.png

Falador Easy: https://i.ivr.fi/runelite/3d23b8fe.png
Falador Medium: https://i.ivr.fi/runelite/2b56668f5ee7.png
Falador Medium: https://i.ivr.fi/runelite/35f45010.png

Fremennik Easy: https://i.ivr.fi/runelite/2f45244c.png
Fremennik Medium: https://i.ivr.fi/runelite/3ee160ba.png

Kandarin Hard: https://i.ivr.fi/runelite/917deca6.png
Kandarin Medium: https://i.ivr.fi/runelite/320dbfc45970.png

Karamja Easy: https://i.ivr.fi/runelite/1fa958ac.png
Karamja Medium: https://i.ivr.fi/runelite/266bfe2919a9.png

Kourend & Kebos Easy: https://i.ivr.fi/runelite/a72a083d.png
Kourend & Kebos Medium: https://i.ivr.fi/runelite/35714a837b36.png
Kourend & Kebos Hard: https://i.ivr.fi/runelite/301bba81.png

Lumbrdige & Draynor Medium: https://i.ivr.fi/runelite/603cdbab.png
Lumbridge & Draynor Hard: https://i.ivr.fi/runelite/6ba961086355.png
Lumbridge & Draynor Medium: https://i.ivr.fi/runelite/080dedfbb45d.png
Lumbridge & Draynor Medium: https://i.ivr.fi/runelite/2b59b77a96a3.png

Morytania Easy: https://i.ivr.fi/runelite/19a2cbea.png
Morytania Easy: https://i.ivr.fi/runelite/b4fd1a131161.png
Morytania Elite: https://i.ivr.fi/runelite/25af44ff.png
Morytania Elite: https://i.ivr.fi/runelite/921cf145.png
Morytania Hard: https://i.ivr.fi/runelite/0ba8b68e.png
Morytania Medium: https://i.ivr.fi/runelite/9ef3c630.png

Varrock Hard: https://i.ivr.fi/runelite/dce7f537.png
Varrock Medium: https://i.ivr.fi/runelite/722a7998.png
Varrock Medium: https://i.ivr.fi/runelite/b110fde1.png

Western Province Easy: https://i.ivr.fi/runelite/9c2521f22ab1.png
Western Province Hard: https://i.ivr.fi/runelite/d31b31ebfb3f.png

Wilderness Medium: https://i.ivr.fi/runelite/8019e761524a.png
Wilderness Medium: https://i.ivr.fi/runelite/9e33fe4e.png

**Pull request merged**